.github/workflows/sbom-matrix.yaml

```
@@ -0,0 +1,409 @@
+          composer CycloneDX:make-sbom --output-file="${parent_dir}/${{ matrix.component.composer_output }}" --output-format=JSON --spec-version=1.6
+          ls -la "${parent_dir}/${{ matrix.component.composer_output }}"
+
+      - name: Set up Node.js
+        if: matrix.component.npm_output
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "${{ matrix.component.path }}/package.json"
+          cache: 'npm'
+          cache-dependency-path: "${{ matrix.component.path }}/package-lock.json"
+
+      - name: Install NPM dependencies
+        if: matrix.component.npm_output
+        working-directory: ${{ matrix.component.path }}
+        env:
+          FONTAWESOME_PACKAGE_TOKEN: ${{ secrets.FONTAWESOME_PACKAGE_TOKEN }}
+        run: |
+          if [ "${{ matrix.component.name }}" == "nextcloud" ]; then
+            cd custom-npms/nc-mdi-js && npm ci && cd ../..
+            cd custom-npms/nc-vue-material-design-icons && npm ci && cd ../..
+            cd custom-npms/nc-nextcloud-vue && npm ci && cd ../..
+          fi
+          npm ci
+
+      - name: Generate SBOM (NPM)
+        if: matrix.component.npm_output
+        run: |
+          parent_dir=$(pwd)
+          cd "${{ matrix.component.path }}"
+          npx @cyclonedx/cyclonedx-npm --ignore-npm-errors --package-lock-only --output-format JSON --spec-version 1.6 --output-file "${parent_dir}/${{ matrix.component.npm_output }}"
+          ls -la "${parent_dir}/${{ matrix.component.npm_output }}"
+
+      - name: Upload component SBOM artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: sbom-${{ matrix.component.name }}
+          path: |
+            ${{ matrix.component.composer_output }}
+            ${{ matrix.component.npm_output }}
+          if-no-files-found: ignore
+
+  # Job 4: Upload SBOMs to Dependency Track
+  # This job downloads all SBOM artifacts generated by the previous job and uploads
+  # them to a Dependency Track instance for vulnerability scanning and analysis.
+  # The upload only happens for specific branches (ionos-stable and the feature branch).
+  #
+  # The job:
+  # 1. Downloads all SBOM artifacts from the generate-sbom job
+  # 2. Iterates through each component in the matrix
+  # 3. Determines which SBOM file to upload (combined, composer-only, or npm-only)
+  # 4. Uploads each SBOM to the corresponding Dependency Track project
+  # 5. Uses custom CA certificate for secure communication with Dependency Track
+  upload-sboms:
+    needs: [ get-version, setup-matrix, generate-sbom ]
+    runs-on: self-hosted
+
+    steps:
+      - name: Download all SBOM artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: sbom-*
+          merge-multiple: true
+
+      - name: List downloaded files
+        run: |
+          ls -la *.json || echo "No BOM JSON files found"
+
+      - name: Upload SBOMs to Dependency Track
+        if: github.ref == 'refs/heads/ionos-dev'
```

```suggestion
        if: github.ref == 'refs/heads/ionos-stable'
```

let upload for stable branch